sql-plugin/src/main/scala/com/nvidia/spark/rapids/TypeChecks.scala

```diff
@@ -1103,20 +1108,34 @@ object CreateMapCheck extends ExprChecks {
     if (context != ProjectExprContext) {
       meta.willNotWorkOnGpu(s"this is not supported in the $context context")
     } else {
-      if (meta.childExprs.length != 2) {
-        // See https://github.com/NVIDIA/spark-rapids/issues/3229
-        meta.willNotWorkOnGpu("CreateMap only supports two expressions on GPU")
+      // if there are more than two key-value pairs then there is the possibility of duplicate keys
+      if (meta.childExprs.length > 2) {
+        // check for duplicate keys if the keys are literal values
+        val keyExprs = meta.childExprs.indices.filter(_ % 2 == 0).map(meta.childExprs)
+        if (keyExprs.forall(_.wrapped.isInstanceOf[Literal])) {
```

Comment:
Could you use `GpuOverrides.extractLit` instead? It handles cases where ti might be wrapped in an alias, which we have seen before.